internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -669,13 +629,13 @@ func TestReconcile_callAfterClusterUpgrade(t *testing.T) {
 						Spec: clusterv1.ClusterSpec{},
 					},
 					ControlPlane: &scope.ControlPlaneState{
-						Object: controlPlaneStableAtLowerVersion,
+						Object: controlPlaneObj,
 					},
 				},
 				HookResponseTracker: scope.NewHookResponseTracker(),
 				UpgradeTracker: func() *scope.UpgradeTracker {
 					ut := scope.NewUpgradeTracker()
-					ut.ControlPlane.PendingUpgrade = true
+					ut.ControlPlane.IsPendingUpgrade = true
 					return ut
```

Comment:
Please check if it makes sense to add a IsStartingUpgrade case

Reply:
Thinking about it, it could be a valid case. It could be possible that a second upgrade was triggered right when the cluster finished the second upgrade. In that case the reconciler will pick the new upgrade (IsStratingUpgrade) and we should not call the AfterClusterUpgrade hook. 

This case is already covered in our the code, +1 to adding a test case for this as well.